src/Microsoft.Azure.WebJobs.Host/Executors/FunctionExecutor.cs

```diff
@@ -239,16 +239,16 @@ public HostOutputMessage HostOutputMessage
                         traceWriter,
                         instance.FunctionDescriptor);
                     var valueBindingContext = new ValueBindingContext(functionContext, cancellationToken);
-                    var parameters = await instance.BindingSource.BindAsync(valueBindingContext);
-
+                    await parameterHelper.BindAsync(instance.BindingSource, valueBindingContext);
+                    
                     Exception invocationException = null;
                     ExceptionDispatchInfo exceptionInfo = null;
                     string startedMessageId = null;
-                    using (ValueProviderDisposable.Create(parameters))
+                    using (parameterHelper.GetDisposables())
```

Comment:
Yes.  I did this since it was closer to the original code; but I'll take it the next step. 

---
In reply to: [125710090](https://github.com/Azure/azure-webjobs-sdk/pull/1209#discussion_r125710090) [](ancestors = 125710090)